pandapipes/pf/result_extraction.py

```diff
@@ -155,101 +154,117 @@ def extract_branch_results_with_internals(net, branch_results, table_name, res_n
     placement_table = np.argsort(net[table_name].index.values)
     idx_pit = branch_pit[f:t, ELEMENT_IDX]
-    comp_connected = branches_connected[f:t]
+    comp_connected_hyd = get_lookup(net, "branch", "active_hydraulics")[f:t]
+    comp_connected_ht = comp_connected_hyd
+    if consider_heat(mode):
+        comp_connected_ht = get_lookup(net, "branch", "active_heat_transfer")[f:t]
 
     node_pit = net["_pit"]["node"]
 
     # the id of the external node table inside the node_pit (mostly this is "junction": 0)
     ext_node_tbl_idx = get_table_number(get_lookup(net, "node", "table"), node_name)
 
-    if len(res_nodes_from) > 0:
-        # results that relate to the from_node --> in case of many internal nodes, only the single
-        # from_node that is the exterior node (e.g. junction vs. internal pipe_node) result has to
-        # be extracted from the node_pit
-        from_nodes = branch_results["from_nodes"][f:t]
-        from_nodes_external = node_pit[from_nodes, TABLE_IDX_NODE] == ext_node_tbl_idx
-        considered = from_nodes_external & comp_connected
-        external_active = comp_connected[from_nodes_external]
-        for res_name, entry in res_nodes_from:
-            res_table[res_name].values[external_active] = branch_results[entry][f:t][considered]
-
-    if len(res_nodes_to) > 0:
-        # results that relate to the to_node --> in case of many internal nodes, only the single
-        # to_node that is the exterior node (e.g. junction vs. internal pipe_node) result has to
-        # be extracted from the node_pit
-        to_nodes = branch_results["to_nodes"][f:t]
-        to_nodes_external = node_pit[to_nodes, TABLE_IDX_NODE] == ext_node_tbl_idx
-        considered = to_nodes_external & comp_connected
-        external_active = comp_connected[to_nodes_external]
-        for res_name, entry in res_nodes_to:
-            res_table[res_name].values[external_active] = branch_results[entry][f:t][considered]
+    for (res_ext, node_name) in ((res_nodes_from, "from_nodes"), (res_nodes_to, "to_nodes")):
+        if len(res_ext) > 0:
+            # results that relate to the from_node --> in case of many internal nodes, only the
+            # single from_node that is the exterior node (e.g. junction vs. internal pipe_node)
+            # result has to be extracted from the node_pit
+            end_nodes = branch_results[node_name][f:t]
+            end_nodes_external = node_pit[end_nodes, TABLE_IDX_NODE] == ext_node_tbl_idx
+            considered_hyd = end_nodes_external & comp_connected_hyd
+            external_active_hyd = comp_connected_hyd[end_nodes_external]
+            considered_ht, external_active_ht = None, None
+            also_heat = consider_heat(mode, res_ext)
+            if also_heat:
+                considered_ht = end_nodes_external & comp_connected_ht
+                external_active_ht = comp_connected_ht[end_nodes_external]
```

Comment:
I am not really sure what you mean with the version before, because before elements were set oos for both simulations at the same time, we didn't have thermal oos junctions that were in_service in the hydraulics. But it is true that this function is quite nasty. Maybe it is possible to find a solution in combination with the next comment.

Reply:
I implemented a new structure with separation of hydraulic and heat related results and a clearer way to distinguish which results are extracted with which connectivity lookup in which simulation mode (as seen from commit message):

extract hydraulic results with hydraulic connectivity in case of "hydraulics" or "all" simulation mode (not "heat"), extract heat results with hydraulic connectivity in case of "hydraulics" simulation mode and with heat_transfer connectivity in case of "heat" or "all" simulation mode

I hope that helps with the comprehensibility of the code. Here I also addressed the next comment.

Reply:
Another question remains that I posed in another comment: If we only conduct a hydraulic simulation, what should the result for oos junctions wrt temperature look like? We define tfluid_k and could just write it into the result table, but do we want that or rather NaN as the junction is set oos for this simulation?

With the new structure, it remains as tfluid_k, but we can very easily change this behavior by just changing one if-statement in this function.

Reply:
We'll leave it like that for now.